homeassistant/components/satel_integra/binary_sensor.py

```diff
@@ -89,7 +92,7 @@ def name(self):
     @property
     def icon(self):
         """Icon for device by its type."""
-        if self._zone_type == "smoke":
+        if self._zone_type == DEVICE_CLASS_SMOKE:
```

Comment:
Odd that this takes arbitrary strings for the zone type.

Reply:
The zone type is not validated as a device class string. Then we shouldn't compare with our device class constants either.